src/libs/highlevel/elektra.c

```diff
@@ -216,7 +216,7 @@ void elektraSaveKey (Elektra * elektra, Key * key, ElektraError ** error)
 				return;
 			}
 
-			if (elektraKDBErrorCode (kdbSetError) != 30) // ELEKTRA_ERROR_CONFLICT = 30
+			if (elektraKDBErrorCode (kdbSetError) != 03000) // CONFLICT = 03000
```

Comment:
@kodebach, can you help me fix this problem? It is treated as octal and I do not know what to change for your high level api

Reply:
1. Is the error code `03000` used exclusively for the cases were `30` was used before? If not please create a separate error code, otherwise this cannot ever work.
2. Please use the macro instead of the explicit code. The reason we didn't use the macro before is that there was no macro when the code was written.
3. You need to change `elektraKDBErrorCode` to return a `const char *` and use `strcmp` here, now that error codes are strings.

Reply:
Also `doc/decisions/error_message.md` says [Conflict is 02000](https://github.com/ElektraInitiative/libelektra/blob/master/doc/decisions/error_message.md#decision).

Reply:
>1. Is the error code 03000 used exclusively for the cases were 30 was used before? If not please create a separate error code, otherwise this cannot ever work.

Do you really require a completely separate highlevel-only error code for this conflict? Which error from which file do you explicitly want to catch with this codeblock?

Reply:
This code catches the error that has code 30 in the old system. It is used in the resolver, if the storage file is opened exclusively by another process (in which case this code amounts to busy waiting). It is also used, if the file was modified by another process between kdbGet and kdbSet. This normally requires a merge. But the high-level API only changes one key at a time. Therefore we can just repeat the set procedure for the same result.